src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1134,6 +1138,70 @@ public static Collection<FunctionMetadata> ReadFunctionMetadata(ScriptHostConfig
             return functions;
         }
 
+        public Collection<FunctionMetadata> ReadProxyMetadata(ScriptHostConfiguration config, ScriptSettingsManager settingsManager = null)
+        {
+            // read the proxy config
+            string proxyConfigPath = Path.Combine(config.RootScriptPath, ScriptConstants.ProxyMetadataFileName);
+            if (!File.Exists(proxyConfigPath))
+            {
+                return null;
+            }
+
+            settingsManager = settingsManager ?? ScriptSettingsManager.Instance;
+
+            var proxyAppSettingValue = settingsManager.GetSetting(EnvironmentSettingNames.ProxySiteExtensionEnabledKey);
+
+            // This is for backward compatibility only, if the file is present but the value of proxy app setting(ROUTING_EXTENSION_VERSION) is explicitly set to 'disabled' we will ignore loading the proxies.
+            if (!string.IsNullOrWhiteSpace(proxyAppSettingValue) && proxyAppSettingValue.Equals("disabled", StringComparison.OrdinalIgnoreCase))
+            {
+                return null;
+            }
+
+            var proxies = new Collection<FunctionMetadata>();
+            string proxiesJson = File.ReadAllText(proxyConfigPath);
+
+            LoadProxyRoutes(config, proxies, proxiesJson);
+
+            return proxies;
+        }
+
+        private void LoadProxyRoutes(ScriptHostConfiguration config, Collection<FunctionMetadata> proxies, string proxiesJson)
+        {
+            ILoggerFactory loggerFactory = config.HostConfig.LoggerFactory;
+            ILogger proxyStartupLogger = loggerFactory.CreateLogger("Host.Proxies.Initialization");
+
+            if (_proxyClient == null)
+            {
+                _proxyClient = ProxyClientFactory.Create(proxiesJson, proxyStartupLogger);
+            }
+
+            if (_proxyClient == null)
+            {
+                return;
+            }
+
+            var routes = _proxyClient.GetProxyData();
+
+            foreach (var route in routes.Routes)
+            {
+                var proxyMetadata = new FunctionMetadata();
+
+                // This is just to add a http trigger type for proxies so the pipeline doesn't break, the values are all ignored by Proxy.
```

Comment:
Actually, i should get rid of the comment now, after the last iter. We do use these properties. Also, the auth level for proxies is anonymous, currently. We have asks to expose have other auth level, that may come in future.

---
In reply to: [133089031](https://github.com/Azure/azure-webjobs-sdk-script/pull/1749#discussion_r133089031) [](ancestors = 133089031)